docs/source/how-to-cache.mdx

```diff
@@ -0,0 +1,238 @@
+Got 1 error(s) while scanning. Use -vvv to print details.
+```
+
+#### Grep example
+
+Since the output is in tabular format, you can combine it with any `grep`-like tools to
+filter the entries. Here is an example to filter only revisions from the "t5-small"
+model on a Unix-based machine.
+
+
+```text
+➜ eval "huggingface-cli scan-cache -v" | grep "t5-small"
+t5-small                    model     98ffebbb27340ec1b1abd7c45da12c253ee1882a       726.2M        6 refs/pr/1   /Users/lucain/.cache/huggingface/hub/models--t5-small/snapshots/98ffebbb27340ec1b1abd7c45da12c253ee1882a                     
+t5-small                    model     d0a119eedb3718e34c648e594394474cf95e0617       485.8M        6             /Users/lucain/.cache/huggingface/hub/models--t5-small/snapshots/d0a119eedb3718e34c648e594394474cf95e0617                     
+t5-small                    model     d78aea13fa7ecd06c29e3e46195d6341255065d5       970.7M        9 main        /Users/lucain/.cache/huggingface/hub/models--t5-small/snapshots/d78aea13fa7ecd06c29e3e46195d6341255065d5
+```
+
+### From Python
+
+For a more advanced usage, use [`scan_cache_dir`] which is the python utility called by
+the CLI tool.
+
+You can use it to get a detailed report structured around 4 dataclasses:
+- [`HFCacheInfo`]: complete report returned by [`scan_cache_dir`]
+- [`CachedRepoInfo`]: information about a cached repo
+- [`CachedRevisionInfo`]: information about a cached revision (e.g. "snapshot") inside a repo
+- [`CachedFileInfo`]: information about a cached file in a snapshot
+
+Here is a simple usage example. See reference for details.
+
+```py
+>>> from huggingface_hub import scan_cache_dir
+
+>>> hf_cache_info = scan_cache_dir()
+HFCacheInfo(
+    size_on_disk=3398085269,
+    repos=frozenset({
+        CachedRepoInfo(
+            repo_id='t5-small',
+            repo_type='model',
+            repo_path=PosixPath(...),
+            size_on_disk=970726914,
+            nb_files=11,
+            revisions=frozenset({
+                CachedRevisionInfo(
+                    commit_hash='d78aea13fa7ecd06c29e3e46195d6341255065d5',
+                    size_on_disk=970726339,
+                    snapshot_path=PosixPath(...),
+                    files=frozenset({
+                        CachedFileInfo(
+                            file_name='config.json',
+                            size_on_disk=1197
+                            file_path=PosixPath(...),
+                            blob_path=PosixPath(...),
+                        ),
+                        CachedFileInfo(...),
+                        ...
+                    }),
+                ),
+                CachedRevisionInfo(...),
+                ...
+            }),
+        ),
+        CachedRepoInfo(...),
+        ...
+    }),
+    errors=[
+        CorruptedCacheException("Snapshots dir doesn't exist in cached repo: ..."),
+        CorruptedCacheException(...),
+        ...
+    ],
+)
+```
```

Comment:
Yes, agree with you on the need for a utility to delete cached repo/cached revisions. I started to mention this in https://github.com/huggingface/huggingface_hub/pull/990#issuecomment-1226902861 but my plan is definitely to add such a tool. Ideally I would
- merge this PR
- ask on slack if people want to try it before next release (it's already tested but I'd prefer more feedback just because users can have a cache corrupted in a way I did not thought about)
- add the deletion tool and update documentation accordingly
- do all of this before next release so that we ship both the scan and deletion at the same time

About adding an example using `rm` I'm a bit skeptical because revisions have to be carefully deleted (blobs should be deleted only if not referred anymore). Once we have a deletion tool, we can make it work with grep and pipes I think.